File: lakeSide-hotel-demo-client-main/src/model/RoomDetail.js
class RoomDetail {
    constructor(id = null, info = null, photo = null, photo_url = null, room = null) {
        this.id = id;
        this.info = info;
        this.photo = photo; // Đây là Blob ở backend
        this.photo_url = photo_url; // URL của ảnh ở backend
        this.room = room; // Đối tượng của class RoomModel
    }
}

export default RoomDetail;

File: lakeSide-hotel-demo-server-master/src/main/java/com/dailycodework/lakesidehotel/model/RoomDetail.java
package com.dailycodework.lakesidehotel.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import org.apache.commons.lang3.RandomStringUtils;

import java.sql.Blob;
import java.util.ArrayList;
import java.util.List;

/**
 * @author Simpson Alfred
 */
@Entity
@Getter
@Setter
@AllArgsConstructor
public class RoomDetail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private  Long id;
    private String info;
    @Lob
    private Blob photo;
    private String photo_url;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id")
    private Room room;

    public RoomDetail() {};
}
